Toolchain/CMake/CMakeLists.txt

```diff
@@ -236,44 +237,221 @@ if (NOT EXISTS "${NOSMAN_WORKSPACE_DIR}/.nosman/index")
+	set(${out_target_dirs} "${_dep_dirs}" PARENT_SCOPE)
+	set(${out_target_include_dirs} "${_dep_include_dirs}" PARENT_SCOPE)
+endfunction()
+
+function(find_sdk_dependency json out_found_version)
+	# Get the number of sdk dependency entries
+	string(JSON found_dep_version GET "${json}" sdk_dependency)
+
+	set(${out_found_version} "${found_dep_version}" PARENT_SCOPE)
+endfunction()
+
+function(default_cmake_lists nos_plugin_file_path common_deps out_target_name)
+	nos_get_module_info_by_path(${nos_plugin_file_path} plugin_name plugin_version out_json_info)
+
+	message(STATUS "Getting module info from path: ${nos_plugin_file_path}")
+	message(STATUS "Configuring module ${plugin_name}")
+	message(STATUS ${CMAKE_CURRENT_SOURCE_DIR})
+	set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+	set(CMAKE_CXX_STANDARD 20)
+	
+	string(REPLACE "." "_" target_name ${plugin_name})
+
+	find_sdk_dependency(${out_json_info} plugin_sdk_version)
+	if ("${plugin_sdk_version}" STREQUAL "None")
+		message(STATUS "Module is not depended on a SDK version, there is no need for a target")
+		return()
+	endif()
+
+	message(STATUS "Plugin SDK version requested: ${plugin_sdk_version}")
+
+	if (NOT WITH_NODOS_WORKSPACE)
+		message(FATAL_ERROR "This repo currently does not support builds without Nodos workspace. "
+		"Place this repo under nodos-workspace/Module folder and run cmake -S ./Toolchain/CMake -B Build from workspace root.")
+	endif()
+	
+	add_compile_definitions(NOS_DISABLE_DEPRECATED)
+	nos_find_plugin_sdk(${plugin_sdk_version} NOS_PLUGIN_SDK_TARGET NOS_SDK_DIR)
+    if (NOT DEFINED NOS_SDK_DIR)
+        message(FATAL_ERROR "Nodos SDK with version ${plugin_sdk_version} not found, please either install it or choose different version")
+    endif()
+
+	find_all_plugin_dependencies(${out_json_info} found_dependency_targets found_dep_dirs found_include_dirs)
+    list(APPEND INCLUDE_FOLDERS ${CMAKE_CURRENT_SOURCE_DIR} "${CMAKE_CURRENT_SOURCE_DIR}/Include" "${found_include_dirs}")
+
+    nos_generate_flatbuffers("${CMAKE_CURRENT_SOURCE_DIR}/Types" "${CMAKE_CURRENT_SOURCE_DIR}/Include/${target_name}" "cpp" "${NOS_SDK_DIR}/Types;${found_dep_dirs}" ${target_name}_generated)
+    list(APPEND MODULE_DEPENDENCIES_TARGETS ${NOS_PLUGIN_SDK_TARGET} ${target_name}_generated)
+
+	list(APPEND
+		MODULE_DEPENDENCIES_TARGETS
+    	${found_dependency_targets}
+    	${${common_deps}}   # dereference here
+	)
+	message(STATUS "Module dependencies: ${MODULE_DEPENDENCIES_TARGETS}")
+    nos_add_plugin("${target_name}" "${MODULE_DEPENDENCIES_TARGETS}" "${INCLUDE_FOLDERS}")
+	if(TARGET "${target_name}")
+		message(STATUS "Successfully created target: ${target_name}")
+	else()
+		nos_fatal_error("Failed to create target: ${target_name}")
+	endif()
+
+    #Helpers need C++20
+    set_target_properties("${target_name}" PROPERTIES CXX_STANDARD 20)
+	set(${out_target_name} ${target_name} PARENT_SCOPE)
+endfunction()
+
+function(process_new_style_target dir common_dependencies)
```

Comment:
I think we should not use "new" or "old" names in the function names, as new and old are meant to change. It should only state what is it, instead of what it means now.